### PR TITLE
JVM IR: Resolve fake overrides when calculating return types (KT-44867)

### PR DESCRIPTION
**Pull request by sfs**
Note that this PR doesn't fix the problem on the old backend.

### CODE REVIEW

File: compiler/testData/codegen/box/inlineClasses/kt44867.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// WITH_RUNTIME
+// IGNORE_BACKEND: JVM
+
+open class BaseWrapper<T>(val response: T)
+class Wrapper(result: Result<String>) : BaseWrapper<Result<String>>(result)
+
+fun box(): String {
+    return Wrapper(Result.success("OK")).response.getOrThrow())
```
Comment (udalov) [inline compiler/testData/codegen/box/inlineClasses/kt44867.kt:8]: Looks like there's an extra `)` here, which is why this test fails for example for JS, JS_IR and WASM. I wonder why it passes for JVM_IR though...

Comment (sfs) [reply compiler/testData/codegen/box/inlineClasses/kt44867.kt:8]: Fixed. That's truly odd, though.

**Comment by udalov** [comment]
Thank you!

**Comment by udalov** [comment]
Cherry-picked to 1.5.0 in ee7a01e36b7.

**Pull request merged**